### PR TITLE
Unload chunks command

### PR DESCRIPTION
**Pull request by ChargedCreeper**
Unloads chunks that are not in use. I can change the name/label of the command if needed. I have already tested this code, and it does in fact unload chunks.

### CODE REVIEW

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import me.StevenLawson.TotalFreedomMod.TFM_Log;
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.Chunk;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
+@CommandParameters(description = "Super admin command - unloads chunks not currently in use.", usage = "/<command>")
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java]: Don't prefix this with _ Super admin command - _ please.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public class Command_rc extends TFM_Command
+{
+    @Override
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        int numChunks = 0;
+        for(World world : server.getWorlds())
+            numChunks += unloadUnusedChunks(world);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java]: Formatting here.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    
+    private int unloadUnusedChunks(World world)
+    {
+        int numChunks = 0;
+        for(Chunk loadedChunk : world.getLoadedChunks())
+        {
+            if(!world.isChunkInUse(loadedChunk.getX(), loadedChunk.getZ()))
+                if(world.unloadChunk(loadedChunk))
+                    numChunks++;
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java]: More formatting here

**Comment by JeromSar** [comment]
@ChargedCreeper Looks good but needs a quick format. Use source -> format to auto-format your source. :)

**Comment by ChargedCreeper** [comment]
@JeromSar I updated the formatting, and added the braces. The no braces thing is a habit I picked up from a C++ class I took.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+{
+    @Override
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        int numChunks = 0;
+        for (World world : server.getWorlds())
+        {
+            numChunks += unloadUnusedChunks(world);
+        }
+        TFM_Util.adminAction(sender.getName(), "unloading unused chunks.", true);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java]: Trailing full stop here.

**Comment by JeromSar** [comment]
@ChargedCreeper Hey, sorry for the late reply. Just a small tweak, and we can merge this.

**Comment by ChargedCreeper** [comment]
@JeromSar I'm not sure what you mean. What needs to be changed?

**Comment by JeromSar** [comment]
See comment above.

**Comment by JeromSar** [comment]
@ChargedCreeper Any updates on this?

**Comment by ChargedCreeper** [comment]
I updated with what I thought you meant but I'm still not sure.

**Comment by OxLemonxO** [comment]
bump?

**Comment by JeromSar** [comment]
@ChargedCreeper Sorry for the confusion. The `TFM_Util.adminMsg()` message should not end with a full stop and should start with a capital letter. :)

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.Chunk;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
+@CommandParameters(description = "Unloads chunks not currently in use.", usage = "/<command>")
+public class Command_rc extends TFM_Command
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_rc.java]: Probably should be the full command name, and /rc as an alias.

**Comment by ChargedCreeper** [comment]
@JeromSar Ah ok. Fixed it.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_unloadchunks.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+{
+    @Override
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        int numChunks = 0;
+        for (World world : server.getWorlds())
+        {
+            numChunks += unloadUnusedChunks(world);
+        }
+        TFM_Util.adminAction(sender.getName(), "Unloading unused chunks", true);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_unloadchunks.java]: Message should probably be sent before actually unloading chunks. :)

**Comment by JeromSar** [comment]
@ChargedCreeper Thanks for the quick update. One small fix, and I think this PR is ready for merging. :)

**Comment by DUFCLiam** [comment]
@JeromSar That is fixed on my one? If you want to merge that but does not matter if not

**Comment by ChargedCreeper** [comment]
@JeromSar I moved the message broadcast to before the chunks are unloaded.

**Comment by DUFCLiam** [comment]
@ChargedCreeper Thanks for complying I will close my PR now :)

**Comment by JeromSar** [comment]
Thanks for contributing @ChargedCreeper! This PR will be merged soon. :)

**Comment by ChargedCreeper** [comment]
@JeromSar sorry about the delay. I've never seen anything like CLAassistant before and I wanted to be sure it was legit first.

**Comment by JeromSar** [comment]
@ChargedCreeper No problem! :)

**Comment by DUFCLiam** [comment]
Should be closed due to the release of 5.0; @JeromSar

**Comment by Wild1145** [comment]
@JeromSar Bump

**Review by Wild1145 — APPROVED** [review]
This should now be ready to merge.

**Comment by JeromSar** [comment]
Merged in https://github.com/TotalFreedom/TotalFreedomMod/commit/be8203a832b298a7e57120f00bfc8341ca501000

Thanks for contributing!

**Pull request closed without merging**